packages/execute-driver/lib/execute-child.js

```diff
@@ -0,0 +1,134 @@
+import _ from 'lodash';
+import B from 'bluebird';
+import vm from 'vm';
+import { logger } from '@appium/support';
+import { attach } from 'webdriverio';
+
+const log = logger.getLogger('ExecuteDriver Child');
+
+// duplicate defining these keys here so we don't need to re-load a huge appium
+// dependency tree into memory just to run a wdio script
+export const W3C_ELEMENT_KEY = 'element-6066-11e4-a52e-4f735466cecf';
+export const MJSONWP_ELEMENT_KEY = 'ELEMENT';
+
+async function runScript (driverOpts, script, timeout) {
+  if (!_.isNumber(timeout)) {
+    throw new Error('Timeout parameter must be a number');
+  }
+
+  // set up fake logger
+  const logLevels = ['error', 'warn', 'log'];
+  const logs = {};
+  const consoleFns = {};
+  for (const level of logLevels) {
+    logs[level] = [];
+    consoleFns[level] = (...logMsgs) => logs[level].push(...logMsgs);
+  }
+
+  const driver = await attach(driverOpts);
+
+  const fullScript = buildScript(script);
+  // the timeout here will not matter really, but set it anyway to be on the
+  // safe side
+  const vmCtx = vm.runInNewContext(fullScript, {}, {timeout});
```

Comment:
IIRC during my experimentation years ago, the timeout only prevents certain kinds of runaway scripts. like it will prevent an infinite loop but not an async call that is never resolved. or vice versa. i wish i had been more explicit in this comment.

Reply:
i think it's vice versa. but now hopefully moot that we are using vm2